notebook2/notebook/notebook.py

```diff
@@ -275,50 +263,44 @@ def read_conditions(conds):
         "type": maxCond.type}
 
 
-def marshall_notebooks(pods = [], svc_status = None):
-    notebooks = []
+def pod_to_ui_dict(pod, svc_status = None):
+    notebook = {
+        'name': pod.metadata.labels['name'],
+        'pod_name': pod.metadata.name,
+        'svc_name': pod.metadata.labels['svc_name'],
+        'pod_status': pod.status.phase,
+        'creation_date': pod.metadata.creation_timestamp.strftime('%D'),
+        'jupyter_token': pod.metadata.labels['jupyter_token'],
+        'container_status': container_status_for_ui(pod.status.container_statuses),
+        'condition': pod_condition_for_ui(pod.status.conditions),
+        'deletion_timestamp': pod.metadata.deletion_timestamp
+    }
 
-    for pod in pods:
-        notebook = {
-            'name': pod.metadata.labels['name'],
-            'pod_name': pod.metadata.name,
-            'svc_name': pod.metadata.labels['svc_name'],
-            'pod_status': pod.status.phase,
-            'creation_date': pod.metadata.creation_timestamp.strftime('%D'),
-            'jupyter_token': pod.metadata.labels['jupyter_token'],
-            'container_status': read_containers_status(pod.status.container_statuses),
-            'condition': read_conditions(pod.status.conditions)
-        }
+    notebook['url'] = f"/instance/{notebook['svc_name']}/?token={notebook['jupyter_token']}"
 
-        notebook['url'] = f"/instance/{notebook['svc_name']}/?token={notebook['jupyter_token']}"
+    if svc_status is not None:
+        notebook['svc_status'] = svc_status
+    else:
+        notebook['svc_status'] = read_svc_status(notebook['svc_name'])
 
-        if svc_status is not None:
-            notebook['svc_status'] = svc_status
-        else:
-            notebook['svc_status'] = read_svc_status(notebook['svc_name'])
+    return notebook
 
-        notebooks.append(notebook)
-
-    return notebooks
 
+def notebooks_for_ui(pods, svc_status = None):
+    return [pod_to_ui_dict(pod, svc_status) for pod in pods]
 
-def get_live_user_notebooks(user_id = None):
-    if user_id is None:
-        return None
 
+def get_live_user_notebooks(user_id):
     pods = k8s.list_namespaced_pod(
         namespace='default',
         label_selector=f"user_id={user_id}",
-        timeout_seconds=KUBERNETES_TIMEOUT_IN_SECONDS).items
+        _request_timeout=KUBERNETES_TIMEOUT_IN_SECONDS).items
 
     if len(pods) == 0:
-        return None
+        return []
 
     # Kubernetes service will reflect the change immediately, pod will not
-    notebooks = list(filter(lambda n: n['svc_status'] != 'Deleted', marshall_notebooks(pods)))
-
-    if len(notebooks) == 0:
-        return None
+    notebooks = list(filter(lambda n: n['deletion_timestamp'] is None, notebooks_for_ui(pods)))
 
     return notebooks
```

Comment:
We can remove the `if len(pods) == 0` and simplify this to:
```python
pods = k8s...
return [pod_to_ui_dict(pod)
        for pod in pods
        if pod.deletion_timestamp is not None]
```

Reply:
And the comment about services is out of date now.

Reply:
Pods length can be 0

Reply:
addressed the service comment

Reply:
true